configure.ac

```
@@ -1640,30 +1640,33 @@ AC_SUBST(LDFLAGS_NODIST)
 UNIVERSAL_ARCH_FLAGS=
 AC_SUBST(UNIVERSAL_ARCH_FLAGS)
 
+dnl PY_CHECK_CC_WARNING(ENABLE, WARNING)
+AC_DEFUN([PY_CHECK_CC_WARNING], [
+  AC_MSG_CHECKING([if we can $1 $CC $2 warning])
+  AS_VAR_PUSHDEF([py_var], [ac_cv_$1_]m4_normalize($2)[_warning])
+  AS_VAR_COPY([py_cc], [CC])
+  AS_VAR_SET([CC], ["$CC -W$2 -Werror"])
+  AC_CACHE_VAL(
+    [py_var],
+    [AC_COMPILE_IFELSE(
+      [AC_LANG_PROGRAM([[]], [[]])],
+      [AS_VAR_SET([py_var], [yes])],
+      [AS_VAR_SET([py_var], [no])],
+    )]
+  )
+  AS_VAR_COPY([CC], [py_cc])
+  AC_MSG_RESULT([$py_var])
+  AS_VAR_POPDEF([py_var])
+])
+
 # tweak BASECFLAGS based on compiler and platform
 case $GCC in
 yes)
     CFLAGS_NODIST="$CFLAGS_NODIST -std=c99"
 
-    AC_MSG_CHECKING(for -Wextra)
-     ac_save_cc="$CC"
-     CC="$CC -Wextra -Werror"
-     AC_CACHE_VAL(ac_cv_extra_warnings,
-       AC_COMPILE_IFELSE(
-         [
-           AC_LANG_PROGRAM([[]], [[]])
-         ],[
-           ac_cv_extra_warnings=yes
-         ],[
-           ac_cv_extra_warnings=no
-         ]))
-     CC="$ac_save_cc"
-    AC_MSG_RESULT($ac_cv_extra_warnings)
-
-    if test $ac_cv_extra_warnings = yes
-    then
-      CFLAGS_NODIST="$CFLAGS_NODIST -Wextra"
-    fi
+    PY_CHECK_CC_WARNING([enable], [extra])
+    AS_IF([test $ac_cv_enable_extra_warning = yes],
```

You can replace the test with ``AS_VAR_IF()``

```suggestion
    AS_VAR_IF([ac_cv_enable_extra_warning], [yes],
```

---

I'll update the rest